ci/cibw_before_all_windows.sh

```diff
@@ -8,17 +8,33 @@ if [[ "$1" == "" ]] ; then
 fi
 PROJECT_PATH="$1"
 
-# nuget
-nuget install zlib-msvc-x64 -ExcludeVersion -OutputDirectory "$PROJECT_PATH"
-EXTRA_PATH="$PROJECT_PATH\zlib-msvc-x64\build\native\bin_release"
+if [[ "$ARCH" == "ARM64" ]]; then
+    # vcpkg for Windows ARM64
+    VCPKG_ROOT="$PROJECT_PATH/vcpkg"
+    git clone https://github.com/microsoft/vcpkg "$VCPKG_ROOT"
```

Comment:
I think we should pin a specific commit here. Preferably, one that's tagged, and add a comment to specify which tag it corresponds to

```suggestion
    VCPKG_SHA="dd3097e305afa53f7b4312371f62058d2e665320"  # 2025.07.25
    git clone https://github.com/microsoft/vcpkg@"$VCPKG_SHA" "$VCPKG_ROOT"
```

Reply:
better yet, windows images from github come with the `Chocolatey` package manager (https://github.com/actions/runner-images/), would it be possible to use that instead ?

Reply:
@neutrinoceros I am not sure would choco provides packages for win-arm64, but I can check this out from my side and get back on this. Thanks for the suggestion!

Reply:
sounds good. I'll "unresolve" this thread until then, to make sure we don't forget about it. Thank you !

Reply:
@neutrinoceros I just tested this on my local Windows on ARM device. It looks like Chocolatey doesn’t provide zlib binaries for Windows ARM64. However, on x64, zlib installs successfully via Chocolatey, which confirms that zlib packages are not available for Windows ARM64 via Chocolatey. This is because Chocolatey internally relies on NuGet to install zlib.

Reply:
thanks a lot for checking. Can you add this explanation as inline comments so it doesn't get lost ?

Reply:
@neutrinoceros sure!